### DISCUSSION TITLE
Cannot find module 'next-video' or its corresponding type declarations

### DISCUSSION THREAD

**Discussion by haxmart** [muxinc/next-video#140] [Q&A]
Following set up instructions but getting errors on both imports (module and video file)

Marking yes for Typescript when installing, but getting these errors in my editor:

`Cannot find module 'next-video' or its corresponding type declarations`

and

`Cannot find module '@/videos/lesson-1-chapter-1.webm' or its corresponding type declarations`

Unsure if its an error or if the installation instructions need tweaking?

**Comment by luwes** [comment]
thanks for the feedback!

does restarting the TS server in your editor fix it?
any chance you could provide a code sandbox replicating the issue?

I have a working example here https://github.com/luwes/next-video-app
or here https://codesandbox.io/p/devbox/next-video-mux-j9nn73

**Comment by salih-codes** [comment]
@maxhart7. Set moduleResolution to "bundler" in your tsconfig.json file and restart your typescript server. 
```typescript
{
  "compilerOptions": {
    "baseUrl": ".",
    "target": "es5",
    "lib": ["dom", "dom.iterable", "esnext"],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "noEmit": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "bundler",  // <--- Like this
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "incremental": true,
    "paths": {
      "@/*": ["./*"]
    },
    "plugins": [
      {
        "name": "next"
      }
    ]
  },
  "include": [
    "video.d.ts",
    "next-env.d.ts",
    "**/*.ts",
    "**/*.tsx",
    ".next/types/**/*.ts",
    "lint-staged.config.js"
  ],
  "exclude": ["node_modules"]
}

```

**Reply by David05500** [reply]
could you unmark this as an answer please, as this is surely a workaround, and not a good one either.

**Comment by NicHaley** [comment]
I had the same issue and `"moduleResolution": "bundler",` fixed it

**Comment by NicHaley** [comment]
I did get a follow up issue when trying to run my local dev server: `'--moduleResolution' option must be: 'node', 'classic', 'node16', 'nodenext'`.

If anyone else gets this, the `bundler` option is only supported on `typescript` 5.x or higher.

**Comment by travisjbeck** [comment]
What should we do if setting the bundler option breaks something else?

**Reply by MisterMunchkin** [reply]
second this, my build breaks once I change moduleResolution to bundler

**Reply by MohamadrezaKazemian** [reply]
+

**Reply by patrickjmccauley** [reply]
try updating the `compilerOptions` field in tsconfig.json:

```json
    "paths": {
      "your-paths",
+     "next-video": ["node_modules/next-video/dist/components/video.d.ts"]
    },
```

allows for `import Video from 'next-video';` without using bundler, at least on our build

**Comment by David05500** [comment]
AI says to add this to next-video.d.ts, which worked for me.

```
declare module 'next-video' {
  import {
    ComponentProps,
    ForwardRefExoticComponent,
    RefAttributes,
  } from 'react'

  interface VideoProps extends ComponentProps<'video'> {
    src: string
    poster?: string
    blurDataURL?: string
    theme?: any
    as?: any
    transform?: (asset: any) => any
    loader?: (src: string) => string
  }

  const Video: ForwardRefExoticComponent<
    VideoProps & RefAttributes<HTMLVideoElement>
  >
  export default Video
}

```
